### DISCUSSION TITLE
some eurorack panels

### DISCUSSION THREAD

**Discussion by warpeggio** [probonopd/MiniDexed#317] [Show and tell]
Hello friends! I've been having an absolute blast with my prototype build, and wanted to share some progress pics on my rack build. 

These are 3U (Eurorack) panels. I've turned the 1602 display sideways to keep the footprint small. Some further HP could be saved if Minidexed supported a 2- or 3-digit 7segment display -  i could envision a control scheme where only the bank & voice are displayed / changeable, but that's another topic... :D
![20220720_075154](https://user-images.githubusercontent.com/74164317/179995431-1753d312-2246-417f-8340-cc37acfd956a.jpg)
![20220720_075253](https://user-images.githubusercontent.com/74164317/179995447-9bc0a1f9-189b-4479-bef7-6810ed88a377.jpg)
The far right module is a karaoke reverb unit from aliexpress (search "DSP 100 kinds effect") available for about $15. Juanito Moore has a video and schematic for it. It can provide a more "spacey" reverb as well as some chorus and other modulation effects.

The general finish technique here was to use automotive vinyl wrap to cover an aluminum panel. The panel in the middle has a waterslide graphic applied. The middle panel also had the mounting hardware for the display and Raspberry Pi installed PRIOR to vinyl wrapping, which allows the face screws to be somewhat hidden. I'm new to this technique but it shows a lot of promise for custom panel graphics.

These panels will all find their way to a custom case with a beefy 5v supply.

**Comment by probonopd** [comment]
Looks great.

From the looks of it (haven't measured it), once https://github.com/probonopd/MiniDexed/pull/312 is merged you might even be able to fit an OLED in without the need for the rotation.

**Reply by warpeggio** [reply]
BE STILL, MY HEART! That looks great. I'll order a few of those now so i can play with them when it's merged.

**Reply by julesdg6** [reply]
Hi .. I am new here (minidexed) but i was thinking about how cool it would be to have a minidexed with some analogue cv/gate inputs on eurorack panel. I know you can't poly like this, but mono is fine ;) Is there a way maybe that the baremetal build could read inputs from an mpc3008? 4x cv/gate pairs and an OLED... pure fantasy!

**Reply by julesdg6** [reply]
oh my... it appears that "circle" does indeed support i2c, so in theory to add cv ins with an mcp3008, you would only need to add code that reads the i2c values and, i guess, adds a note on/off with the value to midi in.

**Reply by probonopd** [reply]
> oh my... it appears that "circle" does indeed support i2c, so in theory to add cv ins with an mcp3008, you would only need to add code that reads the i2c values and, i guess, adds a note on/off with the value to midi in.

Might even use a Raspberry Pi Pico (attached to the MiniDexed device via serial) with Python to do this, for easier development and more flexitbility.

**Comment by warpeggio** [comment]
@julesdg6 The burden for CV/Gate input won't be the hardware. The behavior is un-defined. The original DX7 is based on a keyboard paradigm - the whole patch is based around a set of envelope generators, working on MULTIPLE CONCURRENT GATES. You could do a CV to MIDI implementation in Minidexed, OR you could use a dedicated CV to MIDI device. But the trick is figuring out **what exactly those CV inputs control.** For example, the Korg Volca FM2 allows you to change the algorithm with its sequencer. Another example might be MIDI control of TG detuning, operator modulation sensitivity, or even operator frequency. To do anything like this requires actual synthesizer architecture, because we're going beyond the original DX7 design. Keep in mind, Minidexed is running 8 tone generators, each equivalent to a full DX7- so how do you pick which TG and Operator within? 

For the sake of conversation, i think it would be overwhelming to simply expose EVERY tone generation parameter via midi. There are simply too many.  And, i'm not sure it's within the Minidexed project scope to incorporate CV to MIDI controls, when those devices exist commercially and in the DIY sphere - it's not unreasonable to leave Pitch CV / Gate as an exercise to the reader. Further, the output is headphone level, but not Eurorack level. Another "Instrument In/Out" module would pair nicely here.

However, should a new discussion topic appear regarding New MIDI Implementation Features, i might have some ideas. :)

**Comment by warpeggio** [comment]
Progress shots. I use a piece of cut up gift card as a cable guide. Just finished smoke testing this module, just need to validate the MIDI, DAC, & rotary, and it will be all done.

![20220903_145730](https://user-images.githubusercontent.com/74164317/188287915-4ca4ffd1-f7bc-4e3e-94aa-99fd4adb7cc1.jpg)
![20220903_150006](https://user-images.githubusercontent.com/74164317/188287917-126fafb0-b192-4f2f-8f0e-ec7b800f7639.jpg)
![20220903_150019](https://user-images.githubusercontent.com/74164317/188287919-602d554a-f333-4497-bf72-ad5efc590656.jpg)
